src/solvers/dgsem_tree/dg_2d.jl

```diff
@@ -38,35 +38,9 @@ end
 # and called from the basic `create_cache` method at the top.
 function create_cache(mesh::Union{TreeMesh{2}, StructuredMesh{2}, UnstructuredMesh2D},
                       equations, volume_integral::VolumeIntegralFluxDifferencing, dg::DG, uEltype)
```

Comment:
It is nice that this general (and better) way to handle the non-conservative terms also reduces the number of `create_cache` functions we need.